prometheus/registry.go

```diff
@@ -194,6 +205,10 @@ func (gf GathererFunc) Gather() ([]*dto.MetricFamily, error) {
 	return gf()
 }
 
+func (gf GathererFunc) HasEscapedCollision() bool {
```

Comment:
this is the most risky change. both mimir and avalanche use gathererfunc

Reply:
nevermind, avalanche doesn't use it

https://github.com/grafana/mimir/blob/main/pkg/api/handlers.go#L279